### PR TITLE
Document possible values for the name attribute of package

### LINKED ISSUE #1260: Please document <package name=/> takes non-names
Hello!  This is a report requested on Matrix.

## Problem description
[The documentation](https://osinside.github.io/kiwi/working_with_kiwi/xml_description.html#the-package-element) does not clearly document that `<package name=""/>` takes things that are not names.

In my case (running on openSUSE/zypper on OBS), `<package name="/usr/bin/dumb-init"/>` worked for me (as long as I have a repo that container a package with `Provides: /usr/bin/dumb-init` in the RPM spec).

## Expected behaviour
Document how to write `<packages/>` elements that pull in dependencies via things other than package names.

## Steps to reproduce the behaviour
Read the [documentation](https://osinside.github.io/kiwi/working_with_kiwi/xml_description.html#the-package-element).

## OS and Software information
+ KIWI version: 9.17.15-lp151.1.1 (on OBS)
+ Operating system: openSUSE Leap 15.1
+ OBS version: 2.11~alpha.20191112T141018.f43864e078 (build.opensuse.org)

## Additional information
My build log, which is probably irrelevant for you:
```
[   69s] [ INFO    ]: 22:14:47 | --> collection type: onlyRequired
[   69s] [ INFO    ]: 22:14:47 | --> package: /usr/bin/dumb-init
[   69s] [ INFO    ]: 22:14:47 | --> package: bosh-job-nats-nats
[   69s] [ INFO    ]: 22:14:47 | --> package: bosh-job-nats-smoke-tests
[   69s] [ DEBUG   ]: 22:14:47 | EXEC: [chroot /usr/src/packages/KIWI-docker/build/image-root zypper --non-interactive --pkg-cache-dir /var/cache/kiwi/packages --reposd-dir /var/cache/kiwi/zypper/repos --solv-cache-dir /var/cache/kiwi/zypper/solv --cache-dir /var/cache/kiwi/zypper --raw-cache-dir /var/cache/kiwi/zypper/raw --config /tmp31q7nu9i install --auto-agree-with-licenses --no-recommends /usr/bin/dumb-init bosh-job-nats-nats bosh-job-nats-smoke-tests]
[   69s] [ INFO    ]: Processing: [                                        ] 0%[ DEBUG   ]: 22:14:47 | system: Loading repository data...
[   70s] [ DEBUG   ]: 22:14:47 | system: Reading installed packages...
[   70s] [ DEBUG   ]: 22:14:48 | system: '/usr/bin/dumb-init' not found in package names. Trying capabilities.
[   70s] [ DEBUG   ]: 22:14:48 | system: Resolving package dependencies...
```

### PR DESCRIPTION
**Pull request by dcermak**
This fixes #1260

Changes proposed in this pull request:
* Document possible values for the name attribute of package

### CODE REVIEW

**Comment by schaefi** [comment]
@tomschr would be great if you can take a look

**Comment by schaefi** [comment]
@dcermak for me this makes perfect sense. Thanks much :+1:

File: doc/source/working_with_kiwi/xml_description.rst [hunk 1]
```diff
@@ -780,7 +780,27 @@ removed), whose name is specified via the mandatory `name` attribute:
    </image>
 
 which adds the package `udev` to the list of packages to be added to the
-initial filesystem.
+initial filesystem. Note, that the value that you pass via the `name`
+attribute is passed directly to the used package manager. Thus, if the
+package manager supports other means how packages can be specified, you may
+pass them here too. For example rpm based package managers (like
```
Comment (tomschr) [inline doc/source/working_with_kiwi/xml_description.rst]: "Here", "there" etc. are usually a source of ambiguity. I guess, it refers to the package manager and not the `name` variable. If you can avoid it you could replace "here" with something else. One idea:

```
Thus, if the package manager supports other means how packages can be specified, you may
pass them in this context too.
```

Maybe there are better words. Or you could rephrase the sentence a bit. I'll leave it up to you. :wink:

File: doc/source/working_with_kiwi/xml_description.rst [hunk 2]
```diff
@@ -780,7 +780,27 @@ removed), whose name is specified via the mandatory `name` attribute:
...
+       <!-- snip -->
+       <packages type="bootstrap">
+           <package name="/usr/bin/my-binary"/>
+       </packages>
+   </image>
+
+Whether this works depends on the package manager and on the environment
+that is being used. In the Open Build Service some `Provides` are not
+visible or cannot be properly extracted from the kiwi description and thus
+relying on `Provides` is generally discouraged.
```
Comment (tomschr) [inline doc/source/working_with_kiwi/xml_description.rst]: I would rephrase the sentence:

```
In the Open Build Service, certain `Provides` either are not visible or cannot be properly
extracted from the KIWI description. Therefore, relying on `Provides` is not recommended.
```

**Comment by dcermak** [comment]
@tomschr Fixed!

**Comment by tomschr** [comment]
@dcermak Looks good. Sorry, for bothering again, but there is only the missing comma from https://github.com/OSInside/kiwi/pull/1263#discussion_r346183907 (`For example, RPM based...`) :wink:

**Comment by dcermak** [comment]
> @dcermak Looks good. Sorry, for bothering again, but there is only the missing comma from [#1263 (comment)](https://github.com/OSInside/kiwi/pull/1263#discussion_r346183907) (`For example, RPM based...`) wink

Whoopsie… should be correct now.
